### ISSUE TITLE
[L5.5] A temporary solution for localePicker

### ISSUE THREAD

**Issue by hocopor** [rainlab/translate-plugin#283]
`LocalePicker` still doesn't work correctly without "Force URL schema".

My solution:

1. Update `Plugin.php`,  `boot` function 

```
\Cms\Classes\CmsController::extend(function($controller) {
    $controller->middleware('RainLab\Translate\Middlewares\Locale');
});
```

2. New middleware in `RainLab\Translate\Middlwares\`

```
class Locale
{
    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Closure  $next
     * @return mixed
     */
    public function handle($request, Closure $next)
    {
        $translator = Translator::instance();
        $translator->isConfigured();
    
        if( !$translator->loadLocaleFromRequest() ) {
            $translator->loadLocaleFromSession();
        }
        
        return $next($request);
    }
}
```

3. Update `routes.php`

```
Event::listen('cms.beforeRoute', function() {
        
      foreach( \RainLab\Translate\Models\Locale::listAvailable() as $code => $locale ) {
            
            Route::middleware('web')->prefix($code)->group( function() {
                Route::any('{slug?}', 'Cms\Classes\CmsController@run')->where( 'slug', '(.*)?' );
            } );
            
      }
    
});
```
in `App::before` `Closure`

Tested on custom theme and `Bonjour`
And it works correctly.

I something broke?

build 420 with L5.5

**Comment by LukeTowers** [comment]
Related #282

**Comment by LukeTowers** [comment]
@hocopor Could you propose this in a Pull Request so we can review it in more detail before merging it?

**Comment by hocopor** [comment]
@LukeTowers I tried

**Comment by Teranode** [comment]
Ive tried this too but no change for me

**Comment by hocopor** [comment]
@Teranode Can you do screenshot it? :)

**Comment by Teranode** [comment]
you can test it:
make a page with just a forward slash (the home page basically)

when you have 2 languages the secondary langue uses the page's name as the alias so click the secondary language in the page url it should show the the page's name as the slug.

![](https://image.prntscr.com/image/nLQdfi8BQF6FYHwXLJNPuQ.png)

Espanol (this cannot be left as just a forward slash it will default back to the page name alias)
![](https://image.prntscr.com/image/KH2Tg90ORyCDE-HDFx1B4A.png)

so when I go to the home page change the language to the secondary language:
it will be: `domain.com/home` instead of `domain.com`.

If i try to force the forward slash i get a 404 error

@hocopor

**Comment by Teranode** [comment]
do you see the issue?

**Comment by hocopor** [comment]
@Teranode I don't see the problem here.

 You set the url for the main page, like `/home` and of course following `/` url will return a 404 error.
You can create one more page with `/` for `es` locale url and redirect request to `/home`.

Sorry if i understood you wrong.

**Comment by Teranode** [comment]
no its not that im setting the url to /home

The current url is `/` 

when i alter the es url to `/` it doesn't work, it goes back to `/home`

**Comment by daftspunk** [comment]
Please test 76b798e89d345957802888142a190e5643a5eafe to see if it fixes the issue.

If not create a brand new issue that references this one.

**Comment by daftspunk** [comment]
Assume this is fixed. Fix included in v1.3.3+
